### PR TITLE
BUG: pandas int extension dtypes has no attribute byteorder

### LINKED ISSUE #55069: BUG: pandas int extension dtypes has no attribute byteorder
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas as pd

df = pd.DataFrame({'a': [1]}, dtype=pd.Int8Dtype())
pd.api.interchange.from_dataframe(df.__dataframe__())
```

### Issue Description

I'm trying to use the [dataframe interchange protocol](https://data-apis.org/dataframe-protocol/latest/purpose_and_scope.html) with pandas dataframes holding int extension dtypes, but I can't convert to interchange and back because the extension dtypes don't have a `byteorder` attribute. 

<details>
<summary>Details</summary>

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[7], line 4
      1 import pandas as pd
      3 df = pd.DataFrame({'a': [1]}, dtype=pd.Int8Dtype())
----> 4 pd.api.interchange.from_dataframe(df.__dataframe__())

File ~/anaconda3/envs/ponder-dev/lib/python3.8/site-packages/pandas/core/interchange/from_dataframe.py:54, in from_dataframe(df, allow_copy)
     51 if not hasattr(df, "__dataframe__"):
     52     raise ValueError("`df` does not support __dataframe__")
---> 54 return _from_dataframe(df.__dataframe__(allow_copy=allow_copy))

File ~/anaconda3/envs/ponder-dev/lib/python3.8/site-packages/pandas/core/interchange/from_dataframe.py:75, in _from_dataframe(df, allow_copy)
     73 pandas_dfs = []
     74 for chunk in df.get_chunks():
---> 75     pandas_df = protocol_df_chunk_to_pandas(chunk)
     76     pandas_dfs.append(pandas_df)
     78 if not allow_copy and len(pandas_dfs) > 1:

File ~/anaconda3/envs/ponder-dev/lib/python3.8/site-packages/pandas/core/interchange/from_dataframe.py:116, in protocol_df_chunk_to_pandas(df)
    114     raise ValueError(f"Column {name} is not unique")
    115 col = df.get_column_by_name(name)
--> 116 dtype = col.dtype[0]
    117 if dtype in (
    118     DtypeKind.INT,
    119     DtypeKind.UINT,
    120     DtypeKind.FLOAT,
    121     DtypeKind.BOOL,
    122 ):
    123     columns[name], buf = primitive_column_to_ndarray(col)

File ~/anaconda3/envs/ponder-dev/lib/python3.8/site-packages/pandas/_libs/properties.pyx:36, in pandas._libs.properties.CachedProperty.__get__()

File ~/anaconda3/envs/ponder-dev/lib/python3.8/site-packages/pandas/core/interchange/column.py:126, in PandasColumn.dtype(self)
    124     raise NotImplementedError("Non-string object dtypes are not supported yet")
    125 else:
--> 126     return self._dtype_from_pandasdtype(dtype)

File ~/anaconda3/envs/ponder-dev/lib/python3.8/site-packages/pandas/core/interchange/column.py:141, in PandasColumn._dtype_from_pandasdtype(self, dtype)
    137 if kind is None:
    138     # Not a NumPy dtype. Check if it's a categorical maybe
    139     raise ValueError(f"Data type {dtype} not supported by interchange protocol")
--> 141 return kind, dtype.itemsize * 8, dtype_to_arrow_c_fmt(dtype), dtype.byteorder

AttributeError: 'Int8Dtype' object has no attribute 'byteorder'

</details>

### Expected Behavior

`pd.api.interchange.from_dataframe(df.__dataframe__())` should not raise an error and should give back the original dataframe.

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 0f437949513225922d851e9581723d82120684a6
python           : 3.8.16.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 22.6.0
Version          : Darwin Kernel Version 22.6.0: Wed Jul  5 22:21:56 PDT 2023; root:xnu-8796.141.3~6/RELEASE_X86_64
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 2.0.3
numpy            : 1.24.3
pytz             : 2023.3
dateutil         : 2.8.2
setuptools       : 67.8.0
pip              : 23.2.1
Cython           : None
pytest           : 7.3.1
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : 3.1.2
lxml.etree       : 4.9.3
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.1.2
IPython          : 8.12.0
pandas_datareader: None
bs4              : 4.12.2
bottleneck       : None
brotli           : None
fastparquet      : None
fsspec           : 2023.6.0
gcsfs            : None
matplotlib       : 3.7.1
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 10.0.1
pyreadstat       : None
pyxlsb           : None
s3fs             : 0.4.2
scipy            : 1.10.1
snappy           : None
sqlalchemy       : 2.0.19
tables           : None
tabulate         : 0.9.0
xarray           : None
xlrd             : None
zstandard        : 0.21.0
tzdata           : 2023.3
qtpy             : None
pyqt5            : None

</details>

### PR DESCRIPTION
**Pull request by MarcoGorelli**
- [ ] closes #55069 (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/interchange/column.py [hunk 1]
```diff
@@ -143,6 +146,8 @@ def _dtype_from_pandasdtype(self, dtype) -> tuple[DtypeKind, int, str, str]:
             byteorder = dtype.numpy_dtype.byteorder
         elif isinstance(dtype, DatetimeTZDtype):
             byteorder = dtype.base.byteorder  # type: ignore[union-attr]
+        elif is_extension_array_dtype(dtype):
```
Comment (mroeschke) [inline pandas/core/interchange/column.py:149]: Could you check `BaseMaskedDtype` instead?

File: pandas/tests/interchange/test_impl.py [hunk 2]
```diff
@@ -392,3 +392,13 @@ def test_large_string():
     result = pd.api.interchange.from_dataframe(df.__dataframe__())
     expected = pd.DataFrame({"a": ["x"]}, dtype="object")
     tm.assert_frame_equal(result, expected)
+
+
+@pytest.mark.parametrize("dtype", ["Int8", "Int8[pyarrow]"])
+def test_nullable_integers(dtype: str) -> None:
+    # https://github.com/pandas-dev/pandas/issues/55069
+    pytest.importorskip("pyarrow")
```
Comment (WillAyd) [inline pandas/tests/interchange/test_impl.py:400]: We only need to skip for the second case no? Any reason we don't just use `pytest.param("Int8[pyarrow], marks=td.skip_if_no("pyarrow"))`?

**Review by WillAyd — COMMENTED** [review]
looks good

File: pandas/tests/interchange/test_impl.py (hunk 2 above)
Comment (MarcoGorelli) [reply pandas/tests/interchange/test_impl.py]: because I didn't know you could do that 🙈

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @MarcoGorelli

**Pull request merged**